### PR TITLE
Change log level for OMP warning

### LINKED ISSUE #7112: Turn off OMP_NUM_THREADS warnings?
Can we just turn off the warnings on each ray.init? We can't force everyone to set the environment variable. 

https://github.com/ray-project/ray/blob/3f99be8dad5e0e1abfaede1f25753a0af74f1648/python/ray/__init__.py#L16-L21

### PR DESCRIPTION
**Pull request by simon-mo**
Fixes #7112

### CODE REVIEW

**Comment by ericl** [comment]
I'm fine with this.

File: python/ray/__init__.py [hunk 1]
```diff
@@ -14,7 +14,7 @@
                       "packaged along with Ray).")
 
 if "OMP_NUM_THREADS" not in os.environ:
-    logger.warning("[ray] Forcing OMP_NUM_THREADS=1 to avoid performance "
+    logger.info("[ray] Forcing OMP_NUM_THREADS=1 to avoid performance "
                    "degradation with many workers (issue #6998). You can "
```
Comment (robertnishihara) [inline python/ray/__init__.py]: probably need to change indentation here and below

**Comment by robertnishihara** [comment]
Looks good to me. Need feedback from @pcmoritz and @edoakes before merging.

**Comment by robertnishihara** [comment]
Actually, our default log level is INFO, so maybe this should be DEBUG instead. @ericl is that ok with you? I'd prefer for this warning to not be shown by default.

**Comment by ericl** [comment]
Sure.

**Review by ericl — APPROVED** [review]

**Review by edoakes — COMMENTED** [review]
I agree it's best not to print this every time ray initializes, but we should probably at least have something in the documentation that describes the issue and why we set it to 1. Maybe on this page:
https://ray.readthedocs.io/en/latest/configure.html

**Review by edoakes — APPROVED** [review]
Thanks for the change!

**Pull request merged**